app/src/main/java/com/example/Egida/presentation/ui/SettingFragment.kt

```diff
@@ -0,0 +1,113 @@
+package com.example.Egida.presentation.ui
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.widget.doAfterTextChanged
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import com.example.Egida.activity.MainActivity
+import com.example.Egida.databinding.SettingFragmentBinding
+import com.example.Egida.presentation.viewModel.MainViewModel
+import com.example.Egida.presentation.viewModel.SettingViewModel
+import com.example.Egida.utils.USER_DB
+import com.example.Egida.utils.replaceFragment
+
+class SettingFragment : Fragment() {
+
+    companion object {
+        fun newInstance() = SettingFragment()
+        const val TAG = " SettingFragment"
+    }
+
+    private lateinit var mBinding: SettingFragmentBinding
+    private lateinit var viewModel: SettingViewModel
+    private lateinit var mainViewModel: MainViewModel
+
+    override fun onStart() {
+        super.onStart()
+        (activity as MainActivity).mAppDrawer.disableDrawer()
```

Comment:
This is the worst practice. A better way - wrap activity with an interface smth like  "DrawerController". 
More common way now - use SAME viewModel or SAME repository instance for communication.